File: ReservationSystem/Entities/Product.cs
using ReservationSystem.Entities;
using System;
using System.Collections.Generic;

namespace ReservationSystemBackend.Entities
{
    public class Product : BaseEntity
    {
        public string Name { get; set; }
        public DateTime Created { get; set; } = DateTime.Now;
        public TimeSpan ReservationStartTime { get; set; } = new TimeSpan(14, 0, 0);
        public TimeSpan ReservationEndTime { get; set; } = new TimeSpan(12, 0, 0);
        public string Introduction { get; set; }
        public string City { get; set; }
        public string UnavailableDays { get; set; }
        public ICollection<PriceRow> PriceRows { get; set; }
        public ICollection<Photo> Photos { get; set; }
        public ICollection<Reservation> Reservations { get; set; }
    }
}

File: client/src/app/_models/product.ts
import { Time } from "@angular/common";
import { Photo } from "./photo";
import { PriceRow } from "./priceRow";
import { Reservation } from "./reservation";

export interface Product {
    id: number,
    name: string;
    created: Date;
    reservationStartTime: Time;
    reservationEndTime: Time;
    introduction: string;
    city: string;
    photos: Photo[];
    reservations: Reservation[];
    priceRows: PriceRow[];
}